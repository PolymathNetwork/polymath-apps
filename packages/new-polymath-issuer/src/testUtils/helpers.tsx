import React, { Fragment } from 'react';
// import createSagaMiddleware from 'redux-saga';
import { render } from 'react-testing-library';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { Action, Middleware } from 'redux';
import { set } from 'lodash';
import Web3FakeProvider from 'web3-fake-provider';
import { RootState } from '~/state/reducers/root';

interface RenderTestOpts {
  initialState: RootState;
}

const middlewares: Middleware[] = [];
const mockStore = configureMockStore(middlewares);

const customRender = (
  node: React.ReactElement<any>,
  testOpts: RenderTestOpts,
  ...opts: any[]
) => {
  const initialState = testOpts.initialState;
  const { rerender, ...result } = render(
    <Provider store={mockStore(initialState)}>{node}</Provider>,
    ...opts
  );

  const customRerender = (rerenderNode: React.ReactElement<any>) => {
    return rerender(<Fragment>{rerenderNode}</Fragment>);
  };

  return { rerender: customRerender, ...result };
};

export * from 'react-testing-library';
export { customRender as render };

export class MockedStore {
  public state: RootState;
  public dispatched: Action[] = [];

  constructor() {
    const { reducer: rootReducer } = require('~/state/reducers/root');
    this.state = rootReducer(undefined, {} as any);
    this.dispatched = [];
  }

  public dispatch = (action: Action) => {
    const { reducer: rootReducer } = require('~/state/reducers/root');
    this.state = rootReducer(this.getState(), action);
    this.dispatched.push(action);
  };

  public getState = () => {
    return { ...this.state };
  };

  public setState = (path: string, value: any) => {
    const newState = {
      ...this.state,
    };
    set(newState, path, value);
    this.state = newState;
  };
}

export class MockEthereumProvider extends Web3FakeProvider {
  public networkVersion = '15';
  public async enable() {}
}

export function getGeneratorOutputs(gen: IterableIterator<any>) {
  const results = [];
  let next = gen.next();
  while (!next.done) {
    results.push(next.value);
    next = gen.next();
  }
  return results;
}

export function mockEthereumBrowser() {
  const ethereum = new MockEthereumProvider();
  (global as any).ethereum = ethereum;
}
