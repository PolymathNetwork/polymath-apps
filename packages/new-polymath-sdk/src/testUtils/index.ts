import Web3PromiEvent from 'web3-core-promievent';
import { GenericContract } from '~/LowLevel/types';
import { PostTransactionResolver } from '~/PostTransactionResolver';

export class MockedContract<T extends GenericContract> {
  public autoResolve: boolean;
  public errorMsg?: string;
  public fakeTxOnePromiEvent = new Web3PromiEvent();
  public fakeTxTwoPromiEvent = new Web3PromiEvent();
  public failureTxPromiEvent = new Web3PromiEvent();

  public fakeTxOne = jest.fn(async () => {
    return () => {
      if (this.autoResolve) {
        this.fakeTxOnePromiEvent.resolve();
      }
      return this.fakeTxOnePromiEvent.eventEmitter;
    };
  });
  public fakeTxTwo = jest.fn(async () => {
    return () => {
      if (this.autoResolve) {
        this.fakeTxTwoPromiEvent.resolve();
      }
      return this.fakeTxTwoPromiEvent.eventEmitter;
    };
  });

  public failureTx = jest.fn(async () => {
    return () => {
      const err = this.errorMsg || 'Test error';
      this.failureTxPromiEvent.reject(new Error(err));
      return this.failureTxPromiEvent.eventEmitter;
    };
  });

  constructor({
    autoResolve = true,
    errorMsg,
  }: { autoResolve?: boolean; errorMsg?: string } = {}) {
    this.autoResolve = autoResolve;
    this.errorMsg = errorMsg;
  }
}

export const getMockTransactionSpec = (
  method: (args: any) => Promise<any>,
  args: any,
  resolver = async () => {}
) => ({
  method,
  args,
  postTransactionResolver: new PostTransactionResolver(resolver),
});
