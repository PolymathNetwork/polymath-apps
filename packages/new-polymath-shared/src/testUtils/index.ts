export * from './factories';

const originalWindow = {
  ...window,
};

interface WindowWithEthereum extends Window {
  ethereum: {
    networkVersion: string;
    _metamask?: {
      isApproved: () => Promise<boolean>;
    };
    enable(): Promise<any>;
  };
}

interface WindowWithWeb3 extends Window {
  web3: {
    version: string;
  };
}

interface MockEthereumBrowserArgs {
  support?: 'legacy' | 'modern';
  options?: {
    networkId: number;
    loaded: boolean;
  };
}

export function mockEthereumBrowser({
  support = 'modern',
  options = {
    networkId: 1,
    loaded: true,
  },
}: MockEthereumBrowserArgs) {
  const { networkId, loaded } = options;

  if (support === 'modern') {
    window = (Object.assign(window, {
      ethereum: {
        networkVersion: loaded ? `${networkId}` : undefined,
      },
    }) as any) as WindowWithEthereum;
  }
  if (support === 'legacy') {
    window = Object.assign(window, {
      web3: {
        version: networkId,
      },
    } as any) as WindowWithWeb3;
  }

  return {
    restore: () => {
      window = originalWindow;
    },
    load: () => {
      if (support === 'modern') {
        (window as WindowWithEthereum).ethereum.networkVersion = `${networkId}`;
      }
      if (support === 'legacy') {
        (window as WindowWithWeb3).web3.version = `${networkId}`;
      }
    },
  };
}
