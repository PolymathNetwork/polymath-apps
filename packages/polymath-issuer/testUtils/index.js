import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';

import * as factories from './factories';

const middlewares = [thunk];
export const mockStore = configureMockStore(middlewares);

export { factories };
