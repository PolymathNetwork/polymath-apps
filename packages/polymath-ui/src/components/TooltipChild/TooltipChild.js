import React, { Fragment } from 'react';
// import styled from 'styled-components';
import ReactTooltip from 'react-tooltip';

const TooltipChild = ({ tooltipContent, showTooltip, children, id }) => (
  <Fragment>
    <span data-tip data-for="TooltipChild" data-tip-disable={!showTooltip}>
      {children}
    </span>

    <ReactTooltip id="TooltipChild">
      <span>{tooltipContent}</span>
    </ReactTooltip>
  </Fragment>
);

export default TooltipChild;
