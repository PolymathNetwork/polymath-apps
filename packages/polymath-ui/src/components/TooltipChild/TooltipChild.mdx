---
name: TooltipChild
---

import { Playground, PropsTable } from 'docz';

import Tooltip from './';

# Tooltip

<PropsTable of={Tooltip} />

## Basic usage

<Playground>
  <TooltipChild triggerText="Hover the button!">This is a Tooltip.</Tooltip>
</Playground>
