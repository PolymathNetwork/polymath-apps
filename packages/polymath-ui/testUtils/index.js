import React from 'react';
import { render } from 'react-testing-library';
import { ThemeProvider } from 'styled-components';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import theme from '../src/theme';

const middlewares = [thunk];

const customRender = (node, ...opts) => {
  const { rerender, ...result } = render(
    <ThemeProvider theme={theme}>{node}</ThemeProvider>,
    ...opts
  );

  const customRerender = (rerenderNode, ...rerenderOpts) => {
    return rerender(
      <ThemeProvider theme={theme}>{rerenderNode}</ThemeProvider>,
      ...rerenderOpts
    );
  };

  return { rerender: customRerender, ...result };
};

export const mockStore = configureMockStore(middlewares);
export * from 'react-testing-library';
export { customRender as render };
