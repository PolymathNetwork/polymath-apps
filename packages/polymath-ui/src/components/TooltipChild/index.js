import TooltipChild from './TooltipChild';

export default TooltipChild;
