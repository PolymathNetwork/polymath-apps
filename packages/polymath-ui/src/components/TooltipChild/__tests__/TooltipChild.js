import React from 'react';
import { render } from '../../../../testUtils';

import TooltipChild from '../TooltipChild';

test('renders without crashing', () => {
  const { container } = render(
    <TooltipChild tooltipContent="Hover the item!">
      This is a Tooltip.
    </TooltipChild>
  );

  expect(container.firstChild).toMatchSnapshot();
});
