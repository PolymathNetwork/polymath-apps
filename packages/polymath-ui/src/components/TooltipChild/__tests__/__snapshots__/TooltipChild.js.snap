// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<span
  currentitem="false"
  data-for="TooltipChild"
  data-tip="true"
  data-tip-disable="true"
>
  This is a Tooltip.
</span>
`;
